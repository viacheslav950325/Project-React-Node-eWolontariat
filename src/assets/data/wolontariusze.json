[
    {
    "id": 1,
    "name": "Tomek",
    "surname":  "Wojciechowski",
    "nick": "Tomek z Gór",
    "short_description":    "Chętnie udzielę lekcji gry na ukulele",
    "description":  "Jestem Tomek. Od dziecka pasjonuje mnie instrument jakim jest ukulele. Bardzo lubię dzielić się swoją pasją, więc jeśli chcesz się nauczyć gry na tym wspaniałym instrumencie to pisz śmiało!",
    "image":    "tomekzgor",
    "categories":   ["Korepetycje"],
    "actions": [ "Warszaty dla dzieci z domów dziecka-nauka gry na ukulele"],
    "email":   "t.wojciechowski@vp.pl"
    },
    {
    "id": 2, 
    "name": "Zosia",
    "surname":  "Wierzbowska",
    "nick": "Zosia123",
    "short_description":    "Pomagam w nauce excela, projektuję strony www, udzielam korepetycji z matematyki",
    "description":  "Hej, nazywam się Zosia. Jestem wolontariuszką od 4 lat. Do tej pory brałam udział w wielu wydarzeniach na terenie Warszawy a od początku pandemii udzielam również korepetycji online. ",
    "image" : "zosiaWierzbowska",
    "categories":  [ "excel", "strony www", "korepetycje"],
    "actions":  ["Szukam korepetytora matematyki dla mojej córki (4 klasa)"],
    "email":   "zosia.wierzbowska@wp.pl"   
    },
    {    
    "id": 3,
    "name": "Marek",
    "surname":  "Kowalski",
    "nick": "Kowal",
    "short_description":    "Złota rączka oraz miłośnik zwierząt",
    "description":  "Jestem Marek, bardzo lubię różne prace ręczne. W moim domu naprawiłem już wszystko co się da, więc jeśli masz jakieś stare radio które nie chce już działać albo przeciekający kran- chętnie pomogę. Dodatkowo mam 2 psy które uwielbiam. Podczas spacerów mogę wyprowadzać również twoje czworonogi.",
    "image":"kowal",
    "categories":  [ "wyprowadzanie psów", "złota rączka"],
    "actions":  ["Poszukuję pilnej pomocy przy naprawie kranu w kuchni", "szkolne warszaty z budowania karmników dla ptaków"],
    "email":   "kowal.marek44@gmail.com" 
    },    
    {
    "id": 4,
    "name": "Marlena",
    "surname":"Karczewska",
    "nick": "Marla",
    "short_description":    "Chętnie zaprowadzę lub odbiorę twoje dzieci ze szkoły",
    "description":  "Od 3 lat przebywam na rencie, moje dzieci już dawno zostały odchowane przez co czuję się trochę samotnie w moim domu. Jeśli długo pracujesz chętnie odbiorę twoje dzieci ze szkoły i razem poczekamy na twój powrót z pracy.",
    "image":"marlenaKarczewska",
    "categories":   ["Opieka nad dziećmi"],
    "actions": [null],    
    "email":   "marla456@wp.pl"
    }, 
    {    
    "id": 5,
    "name": "Mateusz",
    "surname":  "Lipowicz",
    "nick": "Mati L.",
    "short_description":    "Pomogę ci przetłumaczyć dokumenty z języka niemieckiego",
    "description":  "Od 15 lat mieszkam w Niemczech, więc bardzo dobrze poznałem język niemiecki. Jeśli masz problem z jakimś tłumaczeniem chętnie pomogę",
    "image":    "mateuszLipowicz",
    "categories":   ["tłumaczenia"],
    "actions":  ["Szukam kogoś do przetłumaczenia ogłoszenia o zaginięciu kotki na język niemiecki"],
    "email":   "mateusz.lipowicz45@wp.pl"
    },    
    {  
    "id": 6,  
    "name": "Justyna",
    "surname":  "Kosińska",
    "nick": "Justyna Kosińska",
    "short_description":"Udzielam korepatycji z chemii",
    "description":  "Na co dzień pracuję w laboratorium chemicznym. Chemia to moja pasja. Prowadziłam już różne warsztaty, m.in. Z tworzenia własnych kosmetyków. Jeśli twoje dziecko ma problem ze zrozumieniem zagadnień chemicznych napisz do mnie",
    "image":    "justynaKosinska",
    "categories":   ["korepetycje"],
    "actions":  ["Szukam kogoś kto przerobi testy szóstoklasisty z moim synem (chemia, matematyka)", "Warsztaty chemiczne dla najmłodszych"],
    "email":   "justyna.kosa5@vp.pl"
    },   
    { 
    "id": 7,   
    "name": "Robert",
    "surname":  "Pawłowski",
    "nick": "Robson",
    "short_description":    "Udzielam korepetycji z fizyki, matematyki. Mogę cię przygotować do egzaminów wstępnych z architektury",
    "description":  "Jestem na 4 roku architektury. Pamiętam ten stres przeg egzaminem ! Jeśli wiesz czego się spodziewać nie jest on już taki straszny ;) Prowadzę krótkie webinaria opisujące proces przygotowania do egzaminu oraz jak wygląda taki egzamin",
    "image":    "robertPawlowski",
    "categories":   ["Korepetycje"],
    "actions":  ["Krótkie webinarium: co trzeba wiedzieć przed egzaminem wstępnym na architekturę?"],
    "email":   "robson@gmail.com"
    },   
    {  
    "id": 8,  
    "name": "Monika",
    "surname":  "Majchrzak",
    "nick": "Monia",
    "short_description":    "Robię sesje zdjęciowe zarówno w plenerze jak i domowym studio",
    "description":  "Zaczynam swoją przygodę z fotografią portretową. Jeśli jesteś początkującą modelką i chcesz rozbudować swoje portfolio, możemy sobie pomóc nawzajem ! ;) Napisz do mnie i umówimy się na sesję zdjęciową",
    "image" :"monikaMajchrzak",
    "categories":   ["fotografia"],
    "actions":  ["Szukam początkującego fotografa do sesji zdjęciowej"],
    "email":   "moniafoto@op.pl"
    } ,  
    {    
    "id": 9,
    "name": "Paulina",
    "surname"   :"Miłańska",
    "nick": "Paulina23",
    "short_description" :"Wyprowadzam psy na dłuuugie spacery",
    "description":  "Uwielbiam zwierzęta. W swoim rodzinnym domu miałam 4 psy, teraz mieszkam w wynajętym mieszkaniu i nie mam warunków do trzmania zwierząt. Bardzo mi brakuje kontaktu z nimi dlatego jeśli nie czujesz się na siłąch na długie spacery a twój czworonożny przyjaciel patrzy na ciebie smutnymi oczami, napisz do mnie ! Chętnie się przejdziemy na długi spacer po lesie",
    "image":    "paulinaMilanska",
    "categories":   ["wyprowadzanie psów"],
    "actions":[null], 
    "email":"paula9222@gmail.com"  
    }
]