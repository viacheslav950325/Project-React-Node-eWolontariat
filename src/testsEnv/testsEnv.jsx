import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { mount, shallow } from "enzyme";
import * as ReactReduxHooks from "react-redux";

export const testingEnvironment = (defaultStoreState, Component) => {
    const defaultStore = configureStore([thunk])(defaultStoreState);
    const useSelectorSpy = jest.spyOn(ReactReduxHooks, "useSelector");
    const useDispatch = jest.spyOn(ReactReduxHooks, "useDispatch");
  
  
    const shallowWrapper = (store = defaultStore, props) => {
      useSelectorSpy.mockImplementation((state) => state(store.getState()));
      useDispatch.mockImplementation(() => store.dispatch);
  
      return shallow(<Component {...props} />);
    };
  
    const mountWrapper = (store = defaultStore, props) => {
      useDispatch.mockRestore();
  
      return mount(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );
    };
    return { defaultStore, shallowWrapper, mountWrapper };
  };