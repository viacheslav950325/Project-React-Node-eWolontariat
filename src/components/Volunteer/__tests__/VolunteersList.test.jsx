import { testingEnvironment } from "../../../testsEnv/testsEnv";
import VolunterList from "../VolunteersList";
import wolontariusze from "../../../assets/data/wolontariusze.json";

const defaultState = {
    volunteers:{
        volunteers:wolontariusze
    }
}

const {shallowWrapper} = testingEnvironment(defaultState,VolunterList)

describe("VolunteersList",()=>{
    it("Should render correctly",()=>{
        const wrapper = shallowWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
