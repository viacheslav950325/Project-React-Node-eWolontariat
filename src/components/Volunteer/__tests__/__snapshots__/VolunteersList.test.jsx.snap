// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolunteersList Should render correctly 1`] = `
<Styled(MuiBox)>
  <WithStyles(ForwardRef(Typography))
    align="center"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
    variant="subtitle2"
  >
    Najpopularniejsze kategorie:
  </WithStyles(ForwardRef(Typography))>
  <Styled(MuiBox)
    display="flex"
    flexWrap="wrap"
    gridColumnGap="2rem"
    id="filtering-buttons"
    justifyContent="center"
    margin="1rem"
    padding="0 0 2rem 0"
  >
    <ForwardRef(CustomButton)
      color="primary"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
      variant="outlined"
    >
      Wszystkie
    </ForwardRef(CustomButton)>
    <ForwardRef(CustomButton)
      color="primary"
      id="Korepetycje"
      key="item-0"
      onClick={[Function]}
      startIcon={<Memo />}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
      variant="contained"
    >
      <WithStyles(ForwardRef(Divider))
        flexItem={true}
        orientation="vertical"
        style={
          Object {
            "backgroundColor": "#eee",
            "marginRight": "10px",
          }
        }
      />
       
      Korepetycje
    </ForwardRef(CustomButton)>
    <ForwardRef(CustomButton)
      color="primary"
      id="Opieka nad dziećmi"
      key="item-1"
      onClick={[Function]}
      startIcon={<Memo />}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
      variant="contained"
    >
      <WithStyles(ForwardRef(Divider))
        flexItem={true}
        orientation="vertical"
        style={
          Object {
            "backgroundColor": "#eee",
            "marginRight": "10px",
          }
        }
      />
       
      Opieka nad dziećmi
    </ForwardRef(CustomButton)>
    <ForwardRef(CustomButton)
      color="secondary"
      id="Strony www"
      key="item-2"
      onClick={[Function]}
      startIcon={<Memo />}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
      variant="contained"
    >
      <WithStyles(ForwardRef(Divider))
        flexItem={true}
        orientation="vertical"
        style={
          Object {
            "backgroundColor": "#eee",
            "marginRight": "10px",
          }
        }
      />
       
      Strony www
    </ForwardRef(CustomButton)>
    <ForwardRef(CustomButton)
      color="primary"
      id="Excel"
      key="item-3"
      onClick={[Function]}
      startIcon={<Memo />}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
      variant="contained"
    >
      <WithStyles(ForwardRef(Divider))
        flexItem={true}
        orientation="vertical"
        style={
          Object {
            "backgroundColor": "#eee",
            "marginRight": "10px",
          }
        }
      />
       
      Excel
    </ForwardRef(CustomButton)>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    flexWrap="wrap"
    justifyContent="space-evenly"
  >
    <ForwardRef(FormControl)
      style={
        Object {
          "bottom": "10px",
        }
      }
      variant="standard"
    >
      <ForwardRef(InputLabel)>
        Wybierz kategorię
      </ForwardRef(InputLabel)>
      <ForwardRef(Select)
        label="Wybierz kategorię"
        onChange={[Function]}
        style={
          Object {
            "fontSize": "1rem",
            "width": "260px",
          }
        }
        value=""
      >
        <ForwardRef(MenuItem)
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value=""
        >
          <em>
            Wszystkie kategorie
          </em>
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Strony www"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Strony www"
        >
          Strony www
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Excel"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Excel"
        >
          Excel
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Korepetycje"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Korepetycje"
        >
          Korepetycje
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Opieka nad dziećmi"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Opieka nad dziećmi"
        >
          Opieka nad dziećmi
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Wyprowadzanie psów"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Wyprowadzanie psów"
        >
          Wyprowadzanie psów
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Fotografia"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Fotografia"
        >
          Fotografia
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Tłumaczenia"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
          value="Tłumaczenia"
        >
          Tłumaczenia
        </ForwardRef(MenuItem)>
      </ForwardRef(Select)>
    </ForwardRef(FormControl)>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    display="flex"
    flexDirection="row"
    flexWrap="wrap"
    justifyContent="center"
    padding="0 1rem 4rem 1rem"
  />
</Styled(MuiBox)>
`;
